src/main/java/com/artipie/asto/cache/CacheControl.java

```diff
@@ -61,16 +67,56 @@
         }
 
         @Override
-        public CompletionStage<Boolean> validate(final Key item) {
-            return this.origin.validate(item);
+        public CompletionStage<Boolean> validate(final Key item,
+            final Supplier<? extends CompletionStage<? extends Content>> supplier) {
+            return this.origin.validate(item, supplier);
         }
     }
 
     /**
      * Validate cached item: checks if cached value can be used or needs to be updated by fresh
      * value.
      * @param item Cached item
+     * @param content Content supplier
      * @return True if cached item can be used, false if needs to be updated
      */
-    CompletionStage<Boolean> validate(Key item);
+    CompletionStage<Boolean> validate(Key item,
+        Supplier<? extends CompletionStage<? extends Content>> content);
```

Comment:
@g4s8 `Supplier<? extends CompletionStage<? extends Content>>` type looks to be a little bit too complex, what do you think about introducing an interface for this type?